src/main/kotlin/me/zeroeightsix/kami/feature/command/NbtCommand.kt

```diff
@@ -0,0 +1,221 @@
+
+    init {
+        opener = Listener({
+            mc.openScreen(screen)
+            KamiMod.EVENT_BUS.unsubscribe(opener)
+        })
+    }
+
+    override fun register(dispatcher: CommandDispatcher<CommandSource>) {
+        dispatcher register rootLiteral("nbt") {
+            literal("look") {
+                does {
+                    val target = mc.crosshairTarget
+                    val tag = when (target?.type) {
+                        HitResult.Type.BLOCK -> {
+                            val entity = mc.world?.getBlockEntity(BlockPos(target.pos))
+                            if (entity == null) {
+                                Error.NO_BLOCK_ENTITY.send(it.source)
+                                null
+                            } else entity.toTag(CompoundTag())
+                        }
+                        HitResult.Type.ENTITY -> mc.targetedEntity?.toTag(CompoundTag())
+                        else -> {
+                            Error.NO_TARGET.send(it.source)
+                            null
+                        }
+                    }
+
+                    open(tag ?: return@does 1)
+                    0
+                }
+            }
+
+            literal("self") {
+                does {
+                    open(mc.player?.toTag(CompoundTag()) ?: return@does 1)
+                    0
+                }
+            }
+
+            literal("hand") {
+                does {
+                    val stack = mc.player?.mainHandStack
+                    if (stack?.isEmpty == true) {
+                        Error.NO_ITEM.send(it.source)
+                        return@does 1
+                    } else {
+                        val tag = stack?.tag
+                            ?: run { Error.NO_ITEM_NBT.send(it.source); return@does 1 }
+                        open(tag)
+                    }
+                    0
+                }
+            }
+        }
+    }
+
+    private fun open(tag: Tag) {
+        screen.tag = tag
+        KamiMod.EVENT_BUS.subscribe(opener)
+    }
+
+    private enum class Error(val message: Text) {
+        NO_TARGET(text(Formatting.RED) { +"No Target found!" }),
+        NO_BLOCK_ENTITY(text(Formatting.RED) { +"This Block is not a BlockEntity!" }),
+        NO_ITEM(text(Formatting.RED) { +"You must hold an item!" }),
+        NO_ITEM_NBT(text(Formatting.RED) { +"The item you are holding has no NBT!" });
+
+        fun send(source: CommandSource) {
+            source replyWith message as MutableText
+        }
+    }
+}
+
+class NbtScreen(
+    var tag: Tag,
+    val defaultColor: Vec4 = Vec4(1f, 1f, 1f, 1f)
+) : InteractableScreen(text(null, "Kami NBT")),
+    widgetsText,
+    widgetsMain,
+    widgetsTrees,
+    contentRegion,
+    childWindows,
+    windowsUtilities,
+    itemWidgetsUtilities,
+    columns,
+    cursorLayout
+// this comment is here to prevent intelli'ntJ from formatting this against the kotlin conventions
```

Comment:
Now this is the proper way to give me a heart attack - we are *not* supposed to be implementing these interfaces!!

`ImGui` implements them, and all of your methods should be using calling `ImGui`'s methods and not `this`.